lib/lib_install.php

```diff
@@ -71,6 +71,104 @@ function checkRequirements(string $dbType = ''): array {
 	);
 }
 
+function getProcessUsername(): string {
+	if (function_exists('posix_getpwuid') && function_exists('posix_geteuid')) {
+		$processUser = posix_getpwuid(posix_geteuid()) ?: [];
+		if (!empty($processUser['name'])) {
+			return $processUser['name'];
+		}
+	}
+
+	if (function_exists('exec')) {
+		exec('whoami', $output);
+		if (!empty($output[0])) {
+			return $output[0];
+		}
+	}
+
+	return _t('install.check.unknown_process_username');
+}
+
+/** @return array{'fileowner':string,'filegroup':string} */
+function getOwnerOfFile(string $file): array {
+	clearstatcache();
+	$stat_file = @stat($file);
+	$owner = [];
+	if(!$stat_file) {
+		//Could’nt stat file
+		$owner['fileowner'] = _t('install.check.unknown_process_username');
+		$owner['filegroup'] = _t('install.check.unknown_process_username');
+		return $owner;
+	}
+	if (function_exists('posix_getpwuid') && function_exists('posix_getgrgid')) {
+		$uid_name = @posix_getpwuid($stat_file['uid']);
+		$gid_name = @posix_getgrgid($stat_file['gid']);
+		$owner['fileowner'] = is_array($uid_name) ? $uid_name['name'] : _t('install.check.unknown_process_username');
+		$owner['filegroup'] = is_array($gid_name) ? $gid_name['name'] : _t('install.check.unknown_process_username');
+		return $owner;
+	};
+	$owner['fileowner'] = 'UID: ' . $stat_file['uid'];
+	$owner['filegroup'] = 'GID: ' . $stat_file['gid'];
+	return $owner;
+}
+
+/**
+ * File permissions (read, write, executable)
+ * Source: https://php.net/function.fileperms
+ */
+function file_permissions(string $path): string {
```

Comment:
I am a bit ambivalent there. While slightly interesting, it is a lot of platform-specific low-level code, which I would rather avoid. It might also be misleading, because we are not checking for all types of errors, especially not on sub-folders/files.

Reply:
`fileperms()` return value will be `false` in any case if it has an error. file_permissions() will be `u` on Windows.

Reply:
How to proceed here? I would like to give information about the `current ` file permission as it is read by the system.
F.e. in https://github.com/FreshRSS/FreshRSS/issues/5134 we can see a lot of unclear information what is the current file permission situation.

Reply:
It is mostly this function, which I do not like, as explained. I will try to find time to think a bit more about it, but I believe this would be better addressed with either:
* a bit more documentation
* some help scripts
* suggest Docker, YunoHost...
* (maybe) try to use system calls such as `exec('ls...')` instead of shipping low-level platform-specific code